### ISSUE TITLE
Expressions: Enable expressions without unified alerting for a single datasource

### ISSUE THREAD

**Issue by bohandley** [grafana/grafana#60340]
We have a request from Azure to enable expressions for a **single datasource** with a feature flag that is separate from the alerting feature flag. I've reached out to the Alerting squad about the feature flag for alerting that includes expressions for datasources. 

These are the results of the discussion

There are two feature flags a customer can set in their custom.ini file

```
[expressions]
enabled = true

[unified_alerting]
enabled = false
```

The outcome of this is that Grafana uses Legacy Alerting but allows expressions to be used in queries.

One caveat is that if a user has already migrated to unified_alerting, and wants to return to legacy alerting, they will need to set `force_migration=true` in their custom.ini file.

See internal discussion here: https://raintank-corp.slack.com/archives/C02UC0KCSF4/p1670954357060059

**Comment by kayodeprinceMS** [comment]
We are looking for separate flags for alerting and expressions in the [plugin.json ](https://grafana.com/docs/grafana/v8.4/developers/plugins/metadata/)

**Comment by bohandley** [comment]
Thank you for the clarification @kayodeprinceMS. @kylebrandt @yesoreyeram Is this possible? Can we add a flag to the plugin.json to enable/disable unified alerting on a single datasource?

**Comment by gillesdemey** [comment]
You can declare a datasource to be compatible with unified alerting, as linked above this can be done with the `plugin.json`'s `alerting` property.

**Comment by kayodeprinceMS** [comment]
As mentioned above, the ask is to be able enable expressions but disable alerting at the plugin level. Is there a separate flag for expression support and alerting support or are they still controlled by the single alerting property?

**Comment by bohandley** [comment]
@grafana/plugins-platform 

> Is there a separate flag for expression support and alerting support or are they still controlled by the single alerting property?

**Comment by tolzhabayev** [comment]
> Is there a separate flag for expression support and alerting support or are they still controlled by the single alerting property?

No, there is currently not ([reference](https://grafana.com/developers/plugin-tools/reference/plugin-json))

**Comment by tolzhabayev** [comment]
PR https://github.com/grafana/grafana/pull/102232
